### PR TITLE
feat: add new 'v2' /delegations endpoints and e2e tests

### PR DESCRIPTION
**Pull request by JoeCap08055**
- Create a `v2` controller `/delegations`
- Change API path from `/delegation` -> `/delegations` to better conform to our REST nomenclature (also reflecting the fact that we can now fetch multiple)
- Add `GET /v2/delegations/:msaId` to get all delegations for an MSA
- Add `GET /v2/delegations/:msaId/:providerId` to get an MSA's delegations to a specific provider
- Use NestJS API versioning to add the `/v1/` and `/v2/` path components instead of hard-coding
- Change response type to not use PolkadotJS SCALE data types; instead create our own DTO so that the generated OpenAPI interface is complete (`/v1` endpoint with PolkadotJS response type was opaque for the response type)
- Update chain setup to add un-delegated and revoked MSAs
- Add E2E tests for new endpoints

# Problem

problem statement, including
Link to GitHub Issue(s):

# Solution

What I/we did to solve this problem

with @pairperson1

## Steps to Verify:

1. A setup step / beginning state
1. What to do next
1. Any other instructions
1. Expected behavior
1. Suggestions for testing

## Screenshots (optional):

Show-n-tell images/animations here

### CODE REVIEW

File: jest.config.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "verbose": true,
+  "silent": true,
```
Comment (mattheworris) [inline jest.config.json:3]: Do `verbose` and `silent` affect different output? Seems like they would be opposites.

Comment (JoeCap08055) [reply jest.config.json:3]: `silent` suppresses console output from the code under test
`verbose` enables detailed test pass/fail reporting output

File: apps/account-api/src/controllers/v1/delegation-v1.controller.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ import {
 } from '@nestjs/common';
 import { ApiBody, ApiCreatedResponse, ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
 
-@Controller('v1/delegation')
-@ApiTags('v1/delegation')
+@Controller({ version: '1', path: 'delegation' })
```
Comment (claireolmstead) [inline apps/account-api/src/controllers/v1/delegation-v1.controller.ts:23]: why is this version a string and the version in the delegation-v2.controller.ts an array?

**Review by mattheworris — APPROVED** [review]
- Read through changes, looks good.

:shipit:

File: apps/account-api/test/delegations.controller.e2e-spec.ts [hunk 3]
```diff
@@ -20,20 +22,25 @@ let publicFollowsSchema: Schema | undefined;
...
+let invalidMsaId: string;
+let msaNonProviderId: string;
 
 describe('Delegation Controller', () => {
   let app: INestApplication;
   let module: TestingModule;
 
   beforeAll(async () => {
-    ({ maxMsaId, provider, users } = await setupProviderAndUsers());
+    ({ maxMsaId, provider, revokedUsers: revokedUser, undelegatedUser, users } = await setupProviderAndUsers());
```
Comment (claireolmstead) [inline apps/account-api/test/delegations.controller.e2e-spec.ts:33]: a little confusing to have a plural -> singular. If a value is plural, I would assume it's an array.

File: jest.config.json (hunk 1 above)
Comment (claireolmstead) [inline jest.config.json:3]: these two are conflicting. what does it do in the config to define them both?

File: libs/account-lib/src/blockchain/blockchain.service.ts [hunk 4]
```diff
@@ -401,8 +403,20 @@ export class BlockchainService implements OnApplicationBootstrap, OnApplicationS
     providerId: AnyNumber,
   ): Promise<CommonPrimitivesMsaDelegation | null> {
     const delegationResponse = await this.api.query.msa.delegatorAndProviderToDelegation(msaId, providerId);
-    if (delegationResponse.isSome) return delegationResponse.unwrap();
-    return null;
+    return delegationResponse.unwrapOr(null);
```
Comment (claireolmstead) [inline libs/account-lib/src/blockchain/blockchain.service.ts:406]: this is cool!

File: apps/account-api/src/controllers/v1/delegation-v1.controller.ts (hunk 2 above)
Comment (JoeCap08055) [reply apps/account-api/src/controllers/v1/delegation-v1.controller.ts:23]: https://github.com/ProjectLibertyLabs/gateway/pull/513/files#r1765252182

**Review by claireolmstead — APPROVED** [review]
LGTM. A couple of questions

File: jest.config.json (hunk 1 above)
Comment (JoeCap08055) [reply jest.config.json:3]: https://github.com/ProjectLibertyLabs/gateway/pull/521#discussion_r1765272223

File: apps/account-api/test/delegations.controller.e2e-spec.ts (hunk 3 above)
Comment (JoeCap08055) [reply apps/account-api/test/delegations.controller.e2e-spec.ts:33]: Good catch; that was a VSCode refactor anomaly. Should be singular; I'll fix.

**Pull request merged**